.github/workflows/analysis.yml

```diff
@@ -1,5 +1,5 @@
 name: code-analysis
-on: [push]
+on: [push, pull_request]
```

Comment:
Yeah, I forgot we kept it to be run only on push earlier. 

Now, I don't think there's any issue in adding pull requests into the scopes. Thanks